src/tests/containerizer/cgroups2_tests.cpp

```diff
@@ -210,6 +221,119 @@ TEST_F(Cgroups2Test, ROOT_CGROUPS2_EnableAndDisable)
   EXPECT_EQ(0u, enabled->count("cpu"));
 }
 
+// Combination of a path and access flags.
+typedef pair<string, int> Access;
+
+using DeviceControllerTestParams = tuple<
+    vector<devices::Entry>, 
+    vector<devices::Entry>,
+    vector<Access>, 
+    vector<Access>>;
+
+class DeviceControllerTestFixture :
+    public Cgroups2Test,
+    public ::testing::WithParamInterface<DeviceControllerTestParams> {};
+
+
+TEST_P(DeviceControllerTestFixture, ROOT_CGROUPS2_DeviceController) {
+  const string& cgroup = TEST_CGROUP;
+
+  auto params = GetParam();
+  const vector<devices::Entry> allow = std::get<0>(params);
+  const vector<devices::Entry> deny = std::get<1>(params);
+  const vector<Access> allowedAccesses = std::get<2>(params);
+  const vector<Access> blockedAccesses = std::get<3>(params);
```

Comment:
these can be refs, since the rhs is not an rvalue reference this pattern definitely copies the vectors, we take the references off when the rhs returns an rvalue reference